### PR TITLE
PD-3739 fixing timeline for new Site traffic report

### PR DESCRIPTION
**Pull request by tejaswinigutti31**
## What's this PR do?
Fix the time line data for new site traffic report
## Where should the reviewer start?
Check the updated python_tld() in myreports/models.py
## How should this be manually tested?
In new site traffic report, check if values in timeline and table below reconcile
Check if the labels displayed are correct on timeline
### Front-End

- [x] Works on IE 11 and Edge
- [x] Works on Mobile

### Separation of Concerns

- [x] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)

### Migrations
N/A

### CODE REVIEW

File: gulp/src/analytics/components/Charts/Line/TimelineChart.jsx [hunk 1]
```diff
@@ -59,6 +59,8 @@ class SimpleLineChart extends Component {
       this.viewType = 'Page Views';
     } else if ( getDataKeys.indexOf('Goals') > -1) {
       this.viewType = 'Goals';
+    } else if ( getDataKeys.indexOf('Apply Clicks') > -1) {
+    this.viewType = 'Apply Clicks';
```
Comment (darrint) [inline gulp/src/analytics/components/Charts/Line/TimelineChart.jsx]: indent

Comment (darrint) [inline gulp/src/analytics/components/Charts/Line/TimelineChart.jsx]: I am not saying we have to fix this now. But:

How can we eliminate this decision from the API? Where in schema dimensions would this "viewType" choice live?

File: myreports/models.py [hunk 2]
```diff
@@ -448,7 +448,10 @@ def regenerate_timeline_data(self, user, company):
         else:
             datasource = self.data_source
         driver = ds_json_drivers[datasource]
-        data_type_name = 'gather_timeline_data'
+        if data_set.value in 'apply_clicks':
```
Comment (darrint) [inline myreports/models.py]: I am not saying we have to fix this now. But:

Where in schema dimensions should this go?

File: myreports/models.py [hunk 3]
```diff
@@ -478,38 +481,51 @@ def python_tld(self):
         report_intention = self.report_schema.split('.', 1)[0]
         if report_intention == 'site-traffic' or report_intention == 'site-traffic-v2':
             if self.data_source == 'goal_events':
-                site_traffic = {
-                   'view_type': 'Goals',
-                   'time_filter': 'time'
-                }
+                if self.report_schema.split('.', 2)[1] in 'apply_clicks':
+                    site_traffic = {
```
Comment (darrint) [inline myreports/models.py]: I am not saying we have to fix this now. But:

Where in schema dimensions should this go?

File: myreports/models.py (hunk 2 above)
Comment (vigowei) [reply myreports/models.py]: is using `in` intended or should this have been an equality comparison?

File: myreports/models.py (hunk 3 above)
Comment (vigowei) [reply myreports/models.py]: one way I can think of is to separate timeline from the regular run_* methods, then set up a similar dispatch for `run_timeline`, or have one `run_timeline_aggregation` method per data_source, that requires metric type(s), for example:
`run_timeline_aggregation(metrics=['visits', 'page_views'], filter_spec=...)` will gather time series data for both visit and page views. 
Now I think it'll be a good opportunity to refactor when we work on the new story of adding multi-dimensions to timeline.

File: myreports/datasources/visits_data.py [hunk 4]
```diff
@@ -257,7 +257,7 @@ def run_gather_timeline_data(self, company, filter_spec, field=None, partial=Non
                                        partial=partial)
         timeline_data = query_set.annotate(
             day=TruncDay('time')
-        ).values('day').order_by('day').annotate(views=Count('site_visit_id'))
+        ).values('day').order_by('day').annotate(visits=Count('site_visit_id'))
```
Comment (vigowei) [inline myreports/datasources/visits_data.py]: this is returning a higher number for timeline in Goal related dimensions, the difference a bit hard to notice, you may not observe it locally if you have good fake data. The delta is caused by the exclusions (`None`/Unavailable) we do for the "regular" goal queries, which is not done for timeline.

**Comment by aljordan82** [comment]
I've tested this PR and I don't see any issues.

**Pull request merged**